sklearn/pipeline.py

```diff
@@ -481,7 +481,7 @@ def _inverse_transform(self, X):
         return Xt
 
     @if_delegate_has_method(delegate='_final_estimator')
-    def score(self, X, y=None):
+    def score(self, X, y=None, **score_params):
```

Comment:
> If we pass **score_params to score function people

Ah wait we are talking about different `score` functions ;)

I suggested

* passing `**score_params` to `self.steps[-1][-1].score(Xt, y, **score_params)` so it doesn't raise an error if `sample_weights` is not supported by the underlying estimator's `score`.
* passing `sample_weights=None` to `Pipeline.score(..., sample_weights=None)` to make it clear that we do not support passing any other parameters, except `sample_weights`.
* documenting at `Pipeline.score` that `sample_weights` can only be passed if the underlying estimator supports it...

I think this is exactly what you suggested already. Sorry for the noise :)